src/animals/dto/animal.dto.ts

```diff
@@ -1,16 +1,95 @@
-import { IsString, IsNotEmpty, IsEnum, IsOptional } from 'class-validator';
+import { Optional } from '@nestjs/common';
+import { ApiProperty, ApiPropertyOptional, PartialType } from '@nestjs/swagger';
+import { Type } from 'class-transformer';
+import {
+  IsString,
+  IsNotEmpty,
+  IsEnum,
+  IsOptional,
+  IsArray,
+  ValidateNested,
+} from 'class-validator';
 import { AnimalType } from 'src/types/animals.type';
 
-export class AnimalDto {
+export class AnimalNameDto {
+  @ApiProperty({
+    description: 'Animal name, can not be empty',
+    type: 'string',
+    example: 'Cat',
+  })
   @IsString()
   @IsNotEmpty()
   animalName: string;
+}
 
+export class AnimalDto extends AnimalNameDto {
+  @ApiProperty({
+    description: `One of Animal types - see Enum`,
+    enum: AnimalType,
+    type: 'AnimalType',
+    example: AnimalType.MAMMALS,
+  })
   @IsNotEmpty()
   @IsEnum(AnimalType)
   type: AnimalType;
 
+  @ApiPropertyOptional({
+    description: 'Optional description',
+    type: 'string',
+    example: 'test description',
+  })
   @IsOptional()
   @IsString()
   description?: string;
 }
+
+export class UpdateAnimalDto extends PartialType(AnimalDto) {}
+
+// export class UpdateAnimalDto {
```

Comment:
?

Reply:
To jest taki sprytny mechanizm, że robi z AnimalDto kopię tylko, że ustawia wszystkie pola jako opcjonalne :)